src/rest_vol.c

```diff
@@ -639,6 +647,58 @@ H5_rest_set_connection_information(void)
      * Attempt to pull in configuration/authentication information from
      * the environment.
      */
+    #ifdef RV_CURL_SOCKET
+    
+      /* This is just a placeholder URL for curl's syntax, its specific value is unimportant */
+      URL = "0";
+      URL_len = 1; 
+
+      if (NULL == (base_URL = (char *) RV_malloc(URL_len + 1)))
+        FUNC_GOTO_ERROR(H5E_VOL, H5E_CANTALLOC, FAIL, "can't allocate space necessary for placeholder base URL");
+    
+      strncpy(base_URL, URL, URL_len);
+      base_URL[URL_len] = '\0';
+
+      /* Get username/password and authenticate as normal in the env case*/
+      const char *username = getenv("HSDS_USERNAME");
+      const char *password = getenv("HSDS_PASSWORD");
+
+    if (username || password) {
```

Comment:
We should probably eventually refactor a bit to share common code between here and below so we aren't duplicating it all.